src/app/trivial_snapp_txn_gen/trivial_snapp_txn_gen.ml

```diff
@@ -0,0 +1,360 @@
+      minWindowDensity: null, 
+      blockchainLength: null, 
+      timestamp: null, 
+      snarkedNextAvailableToken: null, 
+      snarkedLedgerHash: null}, 
+    otherParties: %s, 
+    feePayer: {%s} }})
+}
+    |}
+    other_parties fee_payer
+
+module T = Transaction_snark.Make (struct
+  let constraint_constants = constraint_constants
+
+  let proof_level = proof_level
+end)
+
+let generate_snapp_txn (keypair : Signature_lib.Keypair.t) (ledger : Ledger.t) =
+  let open Deferred.Let_syntax in
+  let receiver =
+    Quickcheck.random_value Signature_lib.Public_key.Compressed.gen
+  in
+  let spec =
+    { Transaction_logic.For_tests.Transaction_spec.sender =
+        (keypair, Account.Nonce.zero)
+    ; fee = Currency.Fee.of_int 10000000000 (*1 Mina*)
+    ; receiver
+    ; amount = Currency.Amount.of_int 10000000000 (*10 Mina*)
+    }
+  in
+  let%bind parties =
+    Transaction_snark.For_tests.create_trivial_predicate_snapp
+      ~constraint_constants spec ledger
+  in
+  Core.printf "Snapp transaction yojson: %s\n\n%!"
+    (Parties.to_yojson parties |> Yojson.Safe.to_string) ;
+  Core.printf "Snapp transaction graphQL input %s\n\n%!"
+    (graphql_snapp_command parties) ;
+  Core.printf "Updated accounts\n" ;
+  List.iter (Ledger.to_list ledger) ~f:(fun acc ->
+      Core.printf "Account: %s\n%!"
+        ( Genesis_ledger_helper_lib.Accounts.Single.of_account acc None
+        |> Runtime_config.Accounts.Single.to_yojson |> Yojson.Safe.to_string )) ;
+  let consensus_constants =
+    Consensus.Constants.create ~constraint_constants
+      ~protocol_constants:Genesis_constants.compiled.protocol
+  in
+  let state_body =
+    let compile_time_genesis =
+      (*not using Precomputed_values.for_unit_test because of dependency cycle*)
+      Mina_state.Genesis_protocol_state.t
+        ~genesis_ledger:Genesis_ledger.(Packed.t for_unit_tests)
+        ~genesis_epoch_data:Consensus.Genesis_epoch_data.for_unit_tests
+        ~constraint_constants ~consensus_constants
+    in
+    compile_time_genesis.data |> Mina_state.Protocol_state.body
+  in
+  let witnesses =
+    Transaction_snark.parties_witnesses_exn ~constraint_constants ~state_body
+      ~fee_excess:Currency.Amount.Signed.zero
+      ~pending_coinbase_init_stack:Pending_coinbase.Stack.empty (`Ledger ledger)
+      [ parties ]
+  in
+  let open Async.Deferred.Let_syntax in
+  let%map _ =
+    Async.Deferred.List.fold ~init:((), ()) (List.rev witnesses)
+      ~f:(fun _ ((witness, spec, statement, snapp_statement) as w) ->
+        Core.printf "%s"
+          (sprintf
+             !"current witness \
+               %{sexp:(Transaction_witness.Parties_segment_witness.t * \
+               Transaction_snark.Parties_segment.Basic.t * \
+               Transaction_snark.Statement.With_sok.t * (int * \
+               Snapp_statement.t)option) }%!"
+             w) ;
+        let%map _ =
+          T.of_parties_segment_exn ~snapp_statement ~statement ~witness ~spec
```

Comment:
Should this (and the instantiation of `T` above) be governed by a flag? I imagine it's not usually useful/desirable to prove the transaction snark for the snapp transaction, except for testing.

Reply:
Ah, yes. I added this to debug. i'll put it behind a flag

Reply:
Will add this in https://github.com/MinaProtocol/mina/pull/9861